alica_engine/src/engine/BasicPlan.cpp

```diff
@@ -67,4 +67,24 @@ void BasicPlan::doTerminate()
     _execContext.store(nullptr);
 }
 
+
+
+void BasicPlan::notifyAssignmentChange(const std::string& newAssignmentName, double oldUtility, double newUtility, size_t numberOfAgents)
+{
+    if (_trace) {
```

Comment:
this check is unnecessary & also probably won't work as expected because its on the main thread. Can remove it.